client/rpc/src/chain/chain_full.rs

```diff
@@ -69,4 +82,54 @@ where
 		let res = self.client.block(&BlockId::Hash(self.unwrap_or_best(hash))).map_err(client_err);
 		async move { res }.boxed()
 	}
+
+	fn block_stats(&self, hash: Option<Block::Hash>) -> Result<Option<BlockStats>> {
+		let block = {
```

Comment:
Your problem is that the block header contains the seal of the consensus engine. The runtime is "dumb" it just compares that the digests of the header that is the result of `execute_block` matches the header of the block you gave as input. The seal is added by the consensus engine after the runtime and is normally stripped before we call into the runtime.

Reply:
Is there anything I can do here? Should I just drop the last `Digest` using `block.header().digest_mut()`?

Reply:
You could drop all of them that are `Digest::Seal`

Reply:
Thanks a lot. I removed all of those before passing them into the runtime and it works now.